flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -439,6 +438,30 @@ public void advanceLastDiscoveredShardOfStream(String stream, String shardId) {
 		}
 	}
 
+	/**
+	 * Function to create new shard consumer (can be overriden to embed customized KinesisProxy).
+	 *
+	 * @param consumerConfigProps config properties of the shard consumer
+	 * @param subscribedShardStateIndex index in subscribed shard state
+	 * @param handle handle of the shard consumer stream
+	 * @param lastSeqNum last processed sequence number of the stream
+	 * @return new shard consumer object
+	 */
+	protected ShardConsumer createShardConsumer(
```

Comment:
There is no need to pass consumerConfigProps to this method. Please see https://github.com/apache/flink/pull/6427/files#diff-0b702244caed9a73b7b3e1dc8a9cbaebR264